src/Grpc.Net.ClientFactory/Internal/GrpcCallInvokerFactory.cs

```diff
@@ -75,12 +77,58 @@ private CallInvoker CreateInvoker(EntryKey key)
         try
         {
             var httpClientFactoryOptions = _httpClientFactoryOptionsMonitor.Get(name);
+            var clientFactoryOptions = _grpcClientFactoryOptionsMonitor.Get(name);
+
+            // gRPC channel is configured with a handler instead of a client, so HttpClientActions aren't used directly.
+            // To capture HttpClient configuration, a temp HttpClient is created and configured using HttpClientActions.
+            // Values from the temp HttpClient are then copied to the gRPC channel.
+            // Only values with overlap on both types are copied so log a message about the limitations.
             if (httpClientFactoryOptions.HttpClientActions.Count > 0)
             {
-                Log.UnsupportedHttpClientActions(_logger, name);
+                Log.HttpClientActionsPartiallySupported(_logger, name);
+
+                var httpClient = new HttpClient(NullHttpMessageHandler.Instance);
+                foreach (var applyOptions in httpClientFactoryOptions.HttpClientActions)
+                {
+                    applyOptions(httpClient);
+                }
+
+                // Copy configuration from HttpClient to GrpcChannel/CallOptions.
+                // This configuration should be overriden by gRPC specific config methods.
+                if (clientFactoryOptions.Address == null)
+                {
+                    clientFactoryOptions.Address = httpClient.BaseAddress;
+                }
+
+                if (httpClient.DefaultRequestHeaders.Any())
+                {
+                    var defaultHeaders = httpClient.DefaultRequestHeaders.ToList();
+
+                    // Merge DefaultRequestHeaders with CallOptions.Headers.
+                    // Follow behavior of DefaultRequestHeaders on HttpClient when merging.
+                    // Don't replace or add new header values if the header name has already been set.
+                    clientFactoryOptions.CallOptionsActions.Add(callOptionsContext =>
+                    {
+                        var metadata = callOptionsContext.CallOptions.Headers ?? new Metadata();
+                        foreach (var entry in defaultHeaders)
+                        {
+                            // grpc requires header names are lower case before being added to collection.
+                            var resolvedKey = entry.Key.ToLower(CultureInfo.InvariantCulture);
+
+                            if (metadata.Get(resolvedKey) == null)
```

Comment:
Would it be interesting to log when it's not null and values form the HttpClient are ignored?

Reply:
I don't think we need to go to this detail.